### ISSUE TITLE
`default_value` and explicit null/nil

### ISSUE THREAD

**Issue by lowip** [absinthe-graphql/absinthe#656]
### Environment

* Elixir version:
```sh
Erlang/OTP 21 [erts-10.2.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]
Elixir 1.7.4 (compiled with Erlang/OTP 21)
```
* Absinthe version (mix deps | grep absinthe):
```sh
* absinthe 1.4.13 (Hex package) (mix)
  locked at 1.4.13 (absinthe) 81eb2ff4
```
* Client Framework and version (Relay, Apollo, etc): NA

### Expected behavior

With the following `input_object`:
```elixir
input_object :comment_filter do
  field :user_id, :id
  field :post_id, :id
  field :parent_id, :id, default_value: nil
end
```
used in the `connection field`:
```elixir
connection field :comments, node_type: :comment do
  arg :filter, non_null(:comment_filter)
  middleware ParseIDs, filter: [user_id: :user, post_id: :post, parent_id: :comment]
  resolve &Resolvers.Core.comments/2
end
```
called with the query:
```graphql
{
  comments(first: 1, filter: {postId: "POST_ID"}) {
    edges { cursor }
  }
}
```
I expect my resolver `&Resolvers.Core.comments/2` to be called with the following args:
```elixir
def comments(args, _res) do
  IO.inspect args
  # %{filter: %{parent_id: nil, post_id: "POST_ID"}, first: 1}
  ...
end
```

### Actual behavior

Instead my args are missing the `nil` default value for `:parent_id`:
```elixir
def comments(args, _res) do
  IO.inspect args
  # %{filter: %{post_id: "POST_ID"}, first: 1}
  ...
end
```

There is currently no way to represent an explicit null/nil value using the `default_value` option.

According to `graphql-js`, when setting `null` as the default value, it should be taken into account:
- https://git.io/fhTT0
- https://git.io/fhTTB

Let me know if this is the expected behaviour for Absinthe, if so I will close the issue.

**Comment by pmargreff** [comment]
I'm working on it.

**Comment by pmargreff** [comment]
Looking into GraphQL language definitions: [DefaultValue can be any Value[Const]](https://spec.graphql.org/draft/#sec-Language.Variables), while [NullValue is a Value[Const]](https://spec.graphql.org/draft/#Value). So your suggestion makes sense, the problem is, taking a brief look into the internals and seems like `nil` value is coupled with value absence. I'll try a POC replacing it with a `%None{}` or something like that.

Suggestions are welcome.

**Comment by pmargreff** [comment]
Put it on hold, to much work on internals for my knowledge level, will work on easier ones first ...

**Comment by binaryseed** [comment]
@benwilson512 Is `default_value: nil` something that should be supported?

**Comment by pmargreff** [comment]
@binaryseed @benwilson512 this feature was implemented in [pmargreff/absinthe/feature/allow-nil-as-default](https://github.com/pmargreff/absinthe/tree/feature/allow-nil-as-default), I was just awaiting the merge of #940 first. 

In the first PR I did the necessary refactor to allow the `default_value: nil` implementation in a simple way. The [branch not submitted already has this feature](https://github.com/pmargreff/absinthe/compare/refactor/extract-steps-from-fill-defaults...pmargreff:feature/allow-nil-as-default) already implemented.

**Comment by sraka1** [comment]
@pmargreff are you still working on this?

**Comment by pmargreff** [comment]
@sraka1 Nope. I have implemented the feature, and put that in a queue to be merged after a refactor. Once the refactor had no response (I closed it after a while) I have never merged the feature itself (both are ref at https://github.com/absinthe-graphql/absinthe/issues/656#issuecomment-735237099).

But it should work flawlessly, as far you rebase [pmargreff/absinthe/feature/allow-nil-as-default](https://github.com/pmargreff/absinthe/tree/feature/allow-nil-as-default) from the current master does not conflict or have any regressions. The new specs at mentioned branch should suffice the intended behavior.

**Comment by benzene73** [comment]
It's still not done. Can't pass in null as default values.
